examples/statemachine/src/language-server/statemachine-formatter.ts

```diff
@@ -0,0 +1,52 @@
+/******************************************************************************
+ * Copyright 2021 TypeFox GmbH
+ * This program and the accompanying materials are made available under the
+ * terms of the MIT License, which is available in the project root.
+ ******************************************************************************/
+
+import type { AstNode } from 'langium';
+import { AbstractFormatter, Formatting } from 'langium';
+import * as ast from './generated/ast.js';
+
+export class StatemachineFormatter extends AbstractFormatter {
+    protected format(node: AstNode): void {
+        if (ast.isState(node)) {
+            const formatter = this.getNodeFormatter(node);
+            formatter.keyword('state')
+                .prepend(Formatting.newLine({ allowMore: true }))
+                .append(Formatting.oneSpace());
+
+            formatter.keyword('actions').append(Formatting.oneSpace());
+            const bracesOpen = formatter.keyword('{');
+            bracesOpen.prepend(Formatting.fit(Formatting.oneSpace(), Formatting.newLine()));
+            const bracesClose = formatter.keyword('}');
+            bracesClose.prepend(Formatting.newLine());
+            formatter.interior(bracesOpen, bracesClose).prepend(Formatting.indent());
+
+            const stateName = formatter.property('name');
+            const stateEnd = formatter.keyword('end');
+            formatter.interior(stateName, stateEnd).prepend(Formatting.indent());
```

Comment:
One more thing, how to doable indent? Currently, the following code:

```
state ExampleState
    actions {
        commandOne
    }
end
```

Would be formatted to:

```
state ExampleState
    actions {
    commandOne
    }
end
```

I tried to do the following:

```ts
formatter.interior(bracesOpen, bracesClose)
    .prepend(Formatting.indent())
    .prepend(Formatting.indent());
```

But it didn't work.

Reply:
I took a look at this, and this seems to be an issue in the formatter. This is probably something we want to fix in a separate PR.

Reply:
I see, I think the outer indent (the one for `state` and `end`) overrides this one.
Or maybe it doesn't take the double indent in consideration.